scripts/ortophotomap_to_patches_dataset.py

```diff
@@ -17,21 +19,42 @@ def geometry_to_pixel_geometry(geometry, transform):
     pix_geom = shp.geometry.Polygon(zip(x_pix, y_pix))
     return pix_geom
 
+
 def load_shapes_df(shapefile_path, transform):
     shapes_df = gpd.read_file(shapefile_path)
-    shapes_df["pixel_geometry"] = shapes_df["geometry"].apply(lambda g: geometry_to_pixel_geometry(g, transform))
+    shapes_df["geometry"] = shapes_df["geometry"].apply(lambda g: g if g.is_valid else g.buffer(0))
+    shapes_df["pixel_geometry"] = shapes_df["geometry"].apply(lambda g: geometry_to_pixel_geometry(g, transform))    
     shapes_df = shapes_df.set_geometry("pixel_geometry")
     shapes_df['pixel_bbox'] = shapes_df["pixel_geometry"].envelope
     shapes_df = shapes_df.merge(shapes_df["pixel_geometry"].bounds.astype(int), left_index=True, right_index=True)
     return shapes_df
 
-def extract_tile(tiff_handler, shapes_df, row_offset, col_offset, tile_size):
+
+def extract_tile(tiff_handler, nir_handler, shapes_df, row_offset, col_offset, tile_size):
     base_window_polygon = shp.geometry.Polygon([[0,0], [0,tile_size], [tile_size,tile_size], [tile_size,0]])
     window_polygon = shp.affinity.translate(base_window_polygon, row_offset, col_offset)
     
     read_window =rio.windows.Window(col_offset, row_offset, tile_size, tile_size)
+    bckg = np.zeros((tile_size, tile_size, 4), dtype=np.uint8)
```

Comment:
Widzę dalej, że jest obsłużone, że nir może być opcjonalny. Jeśli tak to trzebaby dynamicznie zmieniać liczbę kanałów na 3 jeśli `nir_handler is None`

Reply:
4, żeby zostawić alphe, do późniejszego liczenia, nadal się przyda

Reply:
Aaa, no tak :man_facepalming: